### PR TITLE
[prim_rom] Align port names with prim_ram*

### PR DESCRIPTION
**Pull request by msfschaffner**
This aligns port names with the RAM wrappers, which makes is more convenient to generate them from a common template.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

**Comment by imphil** [comment]
We just removed the rst and rvalid_o signals from the RAM primitives (https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim_generic/rtl/prim_generic_ram_1p.sv#L17), maybe we can do the same here to really align?

**Comment by msfschaffner** [comment]
Yeah that makes sense. Shall we make a similar `*_adv.sv` wrapper for the read valid register?

**Comment by msfschaffner** [comment]
Ok, I've split the `rvalid` register into its a `prim_rom_adv` wrapper, similarly as in the SRAM case.

**Comment by msfschaffner** [comment]
Looks like there is still a DV issue related to the `mem_bkdr_if` that I need to resolve (also in #2418)...

**Comment by imphil** [comment]
- On `[prim_ram/rom] Minor restructuring of core files`: Please do not add more files to the catch-all `prim.core`, instead, we need to gradually untangle this one. So instead of adding things there, split them out as done in the prim_ram_1p_adv case.
- On the `mem_bkdr_if`: We had the same problem in https://github.com/lowRISC/opentitan/pull/2299, see the solution there as suggested by Sri.

**Comment by msfschaffner** [comment]
Sure let me amend that. Thanks for the pointer.

File: hw/ip/prim/prim.core [hunk 1]
```diff
@@ -45,6 +47,8 @@ filesets:
       - rtl/prim_subreg.sv
       - rtl/prim_subreg_ext.sv
       - rtl/prim_intr_hw.sv
+      - rtl/prim_rom_adv.sv
+      - rtl/prim_ram_1p_adv.sv
```
Comment (imphil) [inline hw/ip/prim/prim.core:51]: Please revert the change you made to `prim_ram_1p_adv.sv` and leave it a separate file.

File: hw/ip/prim/prim.core [hunk 2]
```diff
@@ -45,6 +47,8 @@ filesets:
       - rtl/prim_subreg.sv
       - rtl/prim_subreg_ext.sv
       - rtl/prim_intr_hw.sv
+      - rtl/prim_rom_adv.sv
```
Comment (imphil) [inline hw/ip/prim/prim.core:50]: Please put this file into a separate core file.

File: hw/top_earlgrey/dv/tb/chip_hier_macros.svh [hunk 3]
```diff
@@ -9,7 +9,7 @@
 `define SPI_DEVICE_HIER `CHIP_HIER.u_spi_device
 `define CPU_HIER        `CHIP_HIER.u_rv_core_ibex
 `define RAM_MAIN_HIER   `CHIP_HIER.u_ram1p_ram_main
-`define ROM_HIER        `CHIP_HIER.u_rom_rom
+`define ROM_HIER        `CHIP_HIER.u_rom_rom.u_prim_rom
```
Comment (imphil) [inline hw/top_earlgrey/dv/tb/chip_hier_macros.svh:12]: That's the approach I had in #2299 initially, but @sriyerg did prefer it the way we have it now. I don't mind either way, but we should be consistent.

File: hw/ip/prim/prim.core (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/prim/prim.core:51]: Sorry shouldn't have force pushed in between changes. I haven't forgotten to do that, but wanted to get CI going in between.

Comment (imphil) [reply hw/ip/prim/prim.core:51]: Ah sorry, it was me being too eager then.

File: hw/top_earlgrey/dv/tb/chip_hier_macros.svh (hunk 3 above)
Comment (msfschaffner) [reply hw/top_earlgrey/dv/tb/chip_hier_macros.svh:12]: Yes, makes sense to be consistent. I'll do it the other way then.

File: hw/top_earlgrey/dv/tb/tb.sv [hunk 4]
```diff
@@ -189,7 +189,7 @@ module tb;
 
     // Backdoors
     uvm_config_db#(virtual mem_bkdr_if)::set(
-        null, "*.env", "mem_bkdr_vifs[Rom]", `ROM_HIER.rom_mem_bkdr_if);
+        null, "*.env", "mem_bkdr_vifs[Rom]", `ROM_HIER.u_prim_rom.rom_mem_bkdr_if);
```
Comment (msfschaffner) [inline hw/top_earlgrey/dv/tb/tb.sv:192]: I've moved this into the testbench now.

**Comment by msfschaffner** [comment]
Ok , I moved the `prim_ram/rom*_adv` sources into their own corefiles. This also affects the `top_earlgrey`, `spi_device` and `usbdev` core files which now need to explicitly include those dependencies.

File: hw/ip/prim/prim_rom_adv.core [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+name: "lowrisc:prim:rom_adv:0.1"
+description: "ROM primitive with advanced features"
+filesets:
+  files_rtl:
+    depend:
+      - lowrisc:prim:rom
```
Comment (imphil) [inline hw/ip/prim/prim_rom_adv.core:11]: Add `- lowrisc:prim:assert` here.

Comment (msfschaffner) [reply hw/ip/prim/prim_rom_adv.core:12]: jep, done

**Comment by msfschaffner** [comment]
Thanks as well!
